test/yorkie_test.ts

```diff
@@ -502,8 +502,17 @@ describe('Yorkie', function () {
 
     const listener1 = new EventEmitter();
     const listener2 = new EventEmitter();
-    const spy1 = createSpy(listener1);
-    const spy2 = createSpy(listener2);
+    const customSpy = (emitter: EventEmitter) => {
+      return (event: ClientEvent | DocEvent) => {
+        if (event.name == ClientEventType.DocumentSyncResult) {
+          emitter.emit(event.value);
+        } else {
+          emitter.emit(event.name);
+        }
+      }
+    }
```

Comment:
Because we need to know that document sync has failed, we need to emit the event's value.
I don't think this looks clean at all, but I wasn't able to come up with a better idea. Please let me know if there's a better way to go around this!